static/monitoring.js

```diff
@@ -109,6 +112,67 @@ function appendLine(element, text) {
     element.appendChild(document.createElement("br"));
 }
 
+function fetchHistoryNextPage() {
+    const next_page = document.getElementsByClassName("page_history_next");
+    const parentdata = next_page.parentElement.dataset;
+    return fetchHistory(next_page.parentElement, ATTESTATION_HISTORY_PAGINATE_NEXT, parentdata.page + 1);
+}
+
+function fetchHistoryPrevPage() {
+    const prev_page = document.getElementsByClassName("page_history_prev");
+    const parentdata = prev_page.parentElement.dataset;
+    return fetchHistory(prev_page.parentElement, ATTESTATION_HISTORY_PAGINATE_BACK, parentdata.page - 1);
+}
+
+function fetchHistory(parent, directive, page=0) {
+    parent.innerHTML = '';
```

Comment:
Can't use `innerHTML` directly with strings because of having trusted types enabled. innerHTML is one of the XSS sinks (can inject arbitrary code at runtime). Rather than writing code to conditionally handle trusted types, it's best to avoid the API and use programmatic DOM manipulation. If the goal is removing all the children, iterate through them and call the method to remove them.

Reply:
gotcha!